src/Simulation/Simulation.jl

```diff
@@ -862,13 +826,22 @@ function _calculate_potential!( sim::Simulation{T, CS}, potential_type::UnionAll
                 (T(1e-5), T(1e-5), T(1e-5))
             end
         end
-        if use_nthreads > Base.Threads.nthreads()
-            use_nthreads = Base.Threads.nthreads();
-            @warn "`use_nthreads` was set to `1`. The environment variable `JULIA_NUM_THREADS` must be set appropriately before the julia session is started."
-        end
         refine = !ismissing(refinement_limits)
         if !(refinement_limits isa Vector) refinement_limits = [refinement_limits] end
         n_refinement_steps = length(refinement_limits)
+
+        max_nthreads, guess_nt = if use_nthreads isa Int 
+            if use_nthreads > Base.Threads.nthreads()
+                use_nthreads = Base.Threads.nthreads();
+                @warn "`use_nthreads` was set to `1`. The environment variable `JULIA_NUM_THREADS` must be set appropriately before the julia session is started."
+            end
+            fill(use_nthreads, n_refinement_steps+1), true
```

Comment:
I guess this will be filled later?

Reply:
`fill(x, n)` creates an vector of length `n` with entries `x`. 

Type stability: `max_nthreads` is otherwise either `Int` or a `Vector{Int}`.
You could still make it type stable but than you produce additional compiled code (methods) for all functions using `max_nthreads`.